benchmarks/src/jmh/java/io/opencensus/benchmarks/trace/StartEndSpanBenchmark.java

```diff
@@ -16,33 +16,49 @@
 
 package io.opencensus.benchmarks.trace;
 
+import io.opencensus.trace.BlankSpan;
 import io.opencensus.trace.Span;
 import io.opencensus.trace.Tracer;
-import io.opencensus.trace.Tracing;
 import io.opencensus.trace.samplers.Samplers;
 import java.util.concurrent.TimeUnit;
 import org.openjdk.jmh.annotations.Benchmark;
 import org.openjdk.jmh.annotations.BenchmarkMode;
 import org.openjdk.jmh.annotations.Mode;
 import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
 import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
 import org.openjdk.jmh.annotations.State;
 import org.openjdk.jmh.annotations.TearDown;
 
 /** Benchmarks for {@link io.opencensus.trace.SpanBuilder} and {@link Span}. */
 @State(Scope.Benchmark)
 public class StartEndSpanBenchmark {
-  private static final Tracer tracer = Tracing.getTracer();
   private static final String SPAN_NAME = "MySpanName";
-  private final Span rootSpan =
-      tracer
-          .spanBuilderWithExplicitParent(SPAN_NAME, null)
-          .setSampler(Samplers.neverSample())
-          .startSpan();
 
-  @TearDown
-  public void doTearDown() {
-    rootSpan.end();
+  @State(Scope.Benchmark)
+  public static class Data {
+    private Tracer tracer;
+    private Span rootSpan = BlankSpan.INSTANCE;
```

Comment:
Shouldn't `rootSpan` always be initialized by the setup method?

Reply:
Yes, but the findbugs framework cannot determine that setup is always called before tear-down so I got an error that tear down may call uninitialized objects.

Reply:
I still think it would be safer to do a null check in the tear down method, or suppress the warning, to avoid the possibility of testing with the wrong span.

Reply:
If we go a null check then we need to do that in all the benchmarks so not ideal. I will add the check that the value is different than the BlankSpan.

Reply:
I don't understand, I thought that FindBugs only required the null check in the tear down method, not in the benchmark methods.  I thought about this more, and I think it might be better to suppress all of the FindBugs nullness warnings, since we are using the Checker Framework now.  The FindBugs null analysis is incomplete, and it sometimes conflicts with the Checker Framework.  I'll work on a PR for that.